src/_blog/rust-libp2p-052-release.md

```diff
@@ -0,0 +1,233 @@
+
+With the `v0.52` release, `rust-libp2p` automatically configures Kademlia in client or server mode depending on our external addresses.
+If we have a confirmed, external address, we will operate in server-mode, otherwise client-mode.
+This is entirely configuration-free (yay!) although follow-up work is under-way to allow setting this manually in certain situations: [#4074](https://github.com/libp2p/rust-libp2p/issues/4074).
+
+We can now do the following:
+
+1. As soon as we learn about an external address (e.g. via AutoNAT), we activate server mode of Kademlia.
+2. Activating server-mode means we allow inbound requests, this is a change in our set of supported protocols.
+3. The change is detected automatically and reported to all protocols as `ConnectionEvent::LocalProtocolsChange`.
+4. `libp2p-identify` picks up this change and pushes it to all connected remote nodes.
+5. Remote nodes can instantly put us into their routing table.
+
+To implement this, several other features/issues had to be fixed.
+If you are interested in the details, read on:
+
+- Changes to the supported protocols are now detected at runtime and communicated to all protocols: [#3651](https://github.com/libp2p/rust-libp2p/pull/3651).
+  
+  Previously, a protocol could retrieve the supported protocols via `PollParameters::supported_protocols`.
+  This list however was computed at start-up and was static.
+  Now, `ConnectionEvent` has two new variants:
+  
+  ```rust
+  pub enum ConnectionEvent<'a> {
+    // existing variants omitted ...
+  
+    /// The local [`ConnectionHandler`] added or removed support for one or more protocols.
+    LocalProtocolsChange(ProtocolsChange<'a>),
+    /// The remote [`ConnectionHandler`] now supports a different set of protocols.
+    RemoteProtocolsChange(ProtocolsChange<'a>),
+  }
+  
+  pub enum ProtocolsChange<'a> {
+    Added(ProtocolsAdded<'a>),
+    Removed(ProtocolsRemoved<'a>),
+  }
+  ``` 
+
+  `ProtocolsAdded` and `ProtocolsRemoved` are iterators over a (new) type called `StreamProtocol`.
+- Protocols are now enforced to be valid UTF-8 strings: [#3746](https://github.com/libp2p/rust-libp2p/pull/3746).
+
+  This was always the case in the specs but the `rust-libp2p` implementation was lagging behind here and improperly represented them as bytes internally.
+- Local changes to our protocols (i.e. a node going from Kademlia server to client mode) are now immediately pushed to the remote via the [`/ipfs/id/push/1.0.0`](https://github.com/libp2p/specs/tree/master/identify#identifypush) protocol: [#3980](https://github.com/libp2p/rust-libp2p/pull/3980).
+- Simplify the scoring mechanism of external addresses: [#3954](https://github.com/libp2p/rust-libp2p/pull/3954)
+
+Not only does this work out-of-the-box and thus improves the developer experience of `rust-libp2p`, it should also result in a much more useful and up-to-date routing table for all users of the Kademlia DHT.
```

Comment:
There are many libp2p Kademlia DHT deployments out there. I find talking of THE Kademlia DHT confusing.

```suggestion
Not only does this work out-of-the-box and thus improves the developer experience of `rust-libp2p`, it should also result in a much more useful and up-to-date routing table for all nodes on a Kademlia DHT.
```